// grommet-controls/components/Card/doc.js
'use strict';

exports.__esModule = true;
exports.docCardContent = exports.docCardSubTitle = exports.docCardTitle = exports.docCardActions = undefined;

var _reactDesc = require('react-desc');

var PAD_SIZES = ['xsmall', 'small', 'medium', 'large'];

var ANIMATION_TYPE = _reactDesc.PropTypes.oneOf(['fadeIn', 'fadeOut', 'jiggle', 'pulse', 'slideUp', 'slideDown', 'slideLeft', 'slideRight', 'zoomIn', 'zoomOut']);

var ANIMATION_SHAPE = _reactDesc.PropTypes.shape({
  type: ANIMATION_TYPE,
  delay: _reactDesc.PropTypes.number,
  duration: _reactDesc.PropTypes.number,
  size: _reactDesc.PropTypes.oneOf(['xsmall', 'small', 'medium', 'large', 'xlarge'])
});

exports.default = function (Element) {
  var DocumentedElement = (0, _reactDesc.describe)(Element).description('A Card-type container, all props of Box apply.').usage('$ npm install grommet-controls\n\n    import { Card } from \'grommet-controls\';\n\n    import { CardTitle, CardSubTitle, CardContent } from \'grommet-controls/components/Card\';\n\n    <Card>\n      <CardTitle>\n        Title\n      </CardTitle>\n      <CardSubTitle>\n        Sub title\n      </CardSubTitle>\n      <CardContent>\n        card content goes here\n      </CardContent>  \n    </Card>  \n');

  DocumentedElement.propTypes = {
    animation: _reactDesc.PropTypes.oneOfType([ANIMATION_TYPE, ANIMATION_SHAPE, _reactDesc.PropTypes.arrayOf(_reactDesc.PropTypes.oneOfType([ANIMATION_TYPE, ANIMATION_SHAPE]))]).description('Animation effect(s) to use. \'duration\' and \'delay\' should\n        be in milliseconds. \'jiggle\' and \'pulse\' types are intended for\n        small elements, like icons.'),
    backContent: _reactDesc.PropTypes.node.description('Content for the back side of the card, if not specified, the card will not be flippable.'),
    background: _reactDesc.PropTypes.oneOfType([_reactDesc.PropTypes.string, _reactDesc.PropTypes.shape({
      color: _reactDesc.PropTypes.string,
      dark: _reactDesc.PropTypes.bool,
      image: _reactDesc.PropTypes.string,
      position: _reactDesc.PropTypes.string,
      opacity: _reactDesc.PropTypes.oneOfType([_reactDesc.PropTypes.oneOf(['weak', 'medium', 'strong']), _reactDesc.PropTypes.bool])
    })]).description('Either a color identifier to use for the background\n        color. For example: \'neutral-1\'. Or, a \'url()\' for an image. Dark\n        is not needed if color is provided.'),
    border: _reactDesc.PropTypes.oneOfType([_reactDesc.PropTypes.oneOf(['top', 'left', 'bottom', 'right', 'horizontal', 'vertical', 'all']), _reactDesc.PropTypes.shape({
      color: _reactDesc.PropTypes.string,
      side: _reactDesc.PropTypes.oneOf(['top', 'left', 'bottom', 'right', 'horizontal', 'vertical', 'all']),
      size: _reactDesc.PropTypes.oneOf(['small', 'medium', 'large'])
    })]).description('Include a border.').defaultValue('all'),
    elevation: _reactDesc.PropTypes.oneOf(['none', 'xsmall', 'small', 'medium', 'large', 'xlarge']).description('Elevated height above the underlying context, indicated\n        via a drop shadow.').defaultValue('small'),
    flipped: _reactDesc.PropTypes.bool.description('Whether the card should be flipped or not (backContent visible).').defaultValue(false),
    flipOnHover: _reactDesc.PropTypes.bool.description('Whether to automatically flip the card content on hover.').defaultValue(true),
    flipDuration: _reactDesc.PropTypes.number.description('Duration in seconds for the flip transition.').defaultValue(0.3),
    gap: _reactDesc.PropTypes.oneOf(['xsmall', 'small', 'medium', 'large', 'xlarge']).description('The amount of spacing between child elements (title, subtitle, content).').defaultValue('small'),
    margin: _reactDesc.PropTypes.oneOfType([_reactDesc.PropTypes.oneOf(['none'].concat(PAD_SIZES)), _reactDesc.PropTypes.shape({
      bottom: _reactDesc.PropTypes.oneOf(PAD_SIZES),
      horizontal: _reactDesc.PropTypes.oneOf(PAD_SIZES),
      left: _reactDesc.PropTypes.oneOf(PAD_SIZES),
      right: _reactDesc.PropTypes.oneOf(PAD_SIZES),
      top: _reactDesc.PropTypes.oneOf(PAD_SIZES),
      vertical: _reactDesc.PropTypes.oneOf(PAD_SIZES)
    })]).description('The amount of margin around the card. An object can\n        be specified to distinguish horizontal margin, vertical margin, and\n        margin on a particular side of the card.').defaultValue('small'),
    onFlip: _reactDesc.PropTypes.func.description('Callback for when the card is flipped.'),
    pad: _reactDesc.PropTypes.oneOfType([_reactDesc.PropTypes.oneOf(['none'].concat(PAD_SIZES)), _reactDesc.PropTypes.shape({
      bottom: _reactDesc.PropTypes.oneOf(PAD_SIZES),
      horizontal: _reactDesc.PropTypes.oneOf(PAD_SIZES),
      left: _reactDesc.PropTypes.oneOf(PAD_SIZES),
      right: _reactDesc.PropTypes.oneOf(PAD_SIZES),
      top: _reactDesc.PropTypes.oneOf(PAD_SIZES),
      vertical: _reactDesc.PropTypes.oneOf(PAD_SIZES)
    })]).description('The amount of padding inside the card content. An\n        object can be specified to distinguish horizontal padding, vertical\n        padding, and padding on a particular side of the box').defaultValue('small'),
    round: _reactDesc.PropTypes.oneOf(['xsmall', 'small', 'medium', 'large', 'full']).description('How much to round the corners of the card.').defaultValue('xsmall'),
    size: _reactDesc.PropTypes.oneOfType([_reactDesc.PropTypes.oneOf(['xxsmall', 'xsmall', 'small', 'medium', 'large', 'xlarge', 'full']), _reactDesc.PropTypes.shape({
      height: _reactDesc.PropTypes.oneOf(['xxsmall', 'xsmall', 'small', 'medium', 'large', 'xlarge', 'full']),
      width: _reactDesc.PropTypes.oneOf(['xxsmall', 'xsmall', 'small', 'medium', 'large', 'xlarge', 'full'])
    })]).description('The size of the Card.').defaultValue({ width: 'large', height: 'medium' })
  };

  return DocumentedElement;
};

var docCardActions = exports.docCardActions = function docCardActions(Element) {
  var DocumentedElement = (0, _reactDesc.describe)(Element).description('Actions container, aligned at bottom of the Card. All properties of Box apply.').usage('$ npm install grommet-controls\n\n    import { CardActions } from \'grommet-controls/components/Card\';\n\n    <Card>\n      <CardActions>\n        <Button ... />\n      </CardActions>\n    </Card>\n');

  DocumentedElement.propTypes = {
    children: _reactDesc.PropTypes.node.description('Content of card actions - usually Button or Anchor elements.').isRequired
  };

  return DocumentedElement;
};

var docCardTitle = exports.docCardTitle = function docCardTitle(Element) {
  var DocumentedElement = (0, _reactDesc.describe)(Element).description('Card title component. All properties of Box and Heading apply.').usage('$ npm install grommet-controls\n\n    import { CardTitle } from \'grommet-controls/components/Card\';\n\n    <Card>\n      <CardTitle>\n        Title\n      </CardTitle>\n    </Card>\n');

  DocumentedElement.propTypes = {
    children: _reactDesc.PropTypes.oneOfType([_reactDesc.PropTypes.string, _reactDesc.PropTypes.node]).description('Content of the title - a string or a react node.').isRequired
  };

  return DocumentedElement;
};

var docCardSubTitle = exports.docCardSubTitle = function docCardSubTitle(Element) {
  var DocumentedElement = (0, _reactDesc.describe)(Element).description('Card sub title component. All properties of Box and Text apply.').usage('$ npm install grommet-controls\n\n    import { CardSubTitle } from \'grommet-controls/components/Card\';\n\n    <Card>\n      <CardSubTitle>\n        Sub title\n      </CardSubTitle>\n    </Card>\n');

  DocumentedElement.propTypes = {
    children: _reactDesc.PropTypes.oneOfType([_reactDesc.PropTypes.string, _reactDesc.PropTypes.node]).description('Content of the sub title - a string or a react node.').isRequired
  };

  return DocumentedElement;
};

var docCardContent = exports.docCardContent = function docCardContent(Element) {
  var DocumentedElement = (0, _reactDesc.describe)(Element).description('Card content component, fills the available space and allows for scrolling if needed. All properties of Box apply.').usage('$ npm install grommet-controls\n\n    import { CardContent } from \'grommet-controls/components/Card\';\n\n    <Card>\n      <CardContent>\n        Some content\n      </CardContent>\n    </Card>\n');

  DocumentedElement.propTypes = {
    children: _reactDesc.PropTypes.node.description('Content of the sub title - a string or a react node.').isRequired
  };

  return DocumentedElement;
};
